consensus/state_test.go

```diff
@@ -1326,20 +1326,79 @@ func TestStartNextHeightCorrectly(t *testing.T) {
 	// add precommits
 	signAddVotes(cs1, types.PrecommitType, nil, types.PartSetHeader{}, vs2)
 	signAddVotes(cs1, types.PrecommitType, theBlockHash, theBlockParts, vs3)
+	time.Sleep(5 * time.Millisecond)
```

Comment:
smelly ... whats this for?

Reply:
To be sure that trigger timeoutPrecommitWait. In case we send all messages at the same time, for some reason it is not deterministic. Not sure why this is the case as it should trigger timeout upon reception of third vote.